aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -34,126 +36,98 @@ void pow_tensor_tensor_kernel(TensorIteratorBase& iter) {
   }
 }
 
-void pow_tensor_scalar_kernel(TensorIteratorBase& iter, const Scalar& exp_scalar) {
-  if (isFloatingType(iter.dtype())) {
-    const auto exp = exp_scalar.to<double>();
-    // Floating types allow AVX2 vector optimizations for pow/sqrt/rsqrt:
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
-      using Vec = Vec256<scalar_t>;
-      if (exp == 0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.sqrt(); }
-        );
-      } else if (exp == 2) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return base * base;
-          },
-          [](Vec base) -> Vec { return base * base; }
-        );
-      } else if (exp == 3) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return base * base * base;
-          },
-          [](Vec base) -> Vec { return base * base * base; }
-        );
-      } else if (exp == -0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
-            return 1.0 / std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.rsqrt(); }
-        );
-      } else if (exp == -1) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return 1.0 / base;
-          },
-          [](Vec base) -> Vec { return base.reciprocal(); }
-        );
-      } else if (exp == -2) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return 1.0 / (base * base);
-          },
-          [](Vec base) -> Vec { return (base * base).reciprocal(); }
-        );
-      } else {
-        cpu_kernel_vec(iter,
-          [=](scalar_t base) -> scalar_t {
-            return std::pow(base, exp);
-          },
-          [=](Vec base) -> Vec { return base.pow(exp); }
-        );
-      }
+// Floating and complex types (minus Half and BFloat16) allow for optimized kernels
+template <typename scalar_t, typename exp_scalar_t>
```

Comment:
Thanks, @heitorschueroff. This is a really good idea & a good learning experience for me. 
I'll keep this cool OOPS design technique/idea in mind.